### DISCUSSION TITLE
No notice changes in the Vercel platform broke our apps and support ignored our tickets

### DISCUSSION THREAD

**Discussion by hades200082** [vercel/community#4610] [Help]
### Summary

We are now two weeks later from the ticket being raised and there hasn't been a single reply from Vercel yet.

**Summary**

Something changed in the Vercel platform on Sept 22nd that caused server-rendered pages on our NextJS 13.1.x apps to constantly time-out, taking a number of our clients' apps offline. 

After upgrading the apps to the latest version of NextJs the issue seems to have vanished and we have seen no further timeouts.

This is deeply troubling from a commercial standpoint and I'm left wondering if we should move our clients' apps to Azure Static Web apps instead now that they support server rendering.

Copy of our ticket below:

---

**11 October 2023 at 08:39**
Vercel appears to treat server-rendered NextJS pages as serverless functions.

Every server-rendered page has been timing out at 60 seconds.

The pages appear to load in their initial state, but the response stream continues until the 60-second mark and then times out, causing client-side code not to mount/hydrate correctly.

According to the usage statistics provided in the Vercel dashboard, this has been happening since Sept 22. Since that date, we have seen 75%+ of function executions timing out.

Before that date, the percentage was less than 8%.

Something clearly changed in Vercel on Sept 22nd and it seems suspicious that it came around the same time as announcements relating to changes in the timeouts for serverless functions, despite the details in the announcements claiming that those changes would not be taking effect until November.

---

**17 October 2023 at 09:09**
We believe that we have resolved this issue by updating to the latest version of NextJS, however, it seems that something changed in the Vercel platform on Sept 22 causing the version of NextJS we were using to begin timing out on app-router-driven server-rendered pages.

We were previously running version 13.1.6. 

After upgrading to the latest version of NextJs the issue seems to have vanished and we have seen no further timeouts. 

To us, it looks like Vercel changed something in the platform, without notice, that effectively took our app offline for the majority of users and then proceeded to ignore our support requests.

This is deeply concerning from a commercial viewpoint.

---

### Steps to Reproduce

Deploy an app-router NextJS 13.1.6 application with server-rendered pages to Vercel. See long page load times and see usage stats indicating serverless function executions timing out.

**Comment by amyegan** [comment]
I'm sorry you're having so much trouble with your Serverless Functions! If you let me know the ticket number, I can check on it for you. 🙇 

With regard to the function timeout change that went into effect for Pro customers on October 1st, a notification was sent ahead of the change and it was added to our changelog at [vercel.com/changelog/serverless-functions-can-now-run-up-to-5-minutes](https://vercel.com/changelog/serverless-functions-can-now-run-up-to-5-minutes). 

The change extended the maximum configurable function duration limit up to 5 minutes for Pro teams and reduced the default to 15 seconds. Previously, the limit and default were both 60 seconds. The newly reduced default number helps reduce the chance for excess Serverless Function Execution usage in the event of external service timeouts or other problems.

For versions of Next.js prior to 13.5, [function duration limits](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) can be configured in a `vercel.json` file with the [`functions`](https://vercel.com/docs/projects/project-configuration#functions) object.

https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration

**Comment by hades200082** [comment]
@amyegan Thanks for the reply, but this issue started happening on Sept 22, well before the function duration limit changes.

Ticket #00164905

As I said in the OP, we eventually fixed the issue for our clients by upgrading their projects to the latest version of NextJS. 

The answers I need now are:

- Why was something changed without notice on Sept 22?
- Why have we had no reply in over 2 weeks? This is unacceptable for any paid hosting platform.
- Given the seemingly complete lack of interest from Vercel in providing any support on this matter, why should we continue recommending Vercel to our clients over other options that also support NextJS such as Netlify or Azure SWA?

I understand that mistakes happen, but when Vercel is being paid to provide a service and that service is broken through no fault of our own, having our support tickets completely ignored is disrespectful and unacceptable.

**Reply by Pieparker** [reply]
Hey @hades200082! I'm Scott, a manager for the support team here at Vercel. Let me give you some of the answers you're looking for and I'll take away some of the questions I'm not yet able to provide good feedback for.

>Why was something changed without notice on Sept 22?

I don't yet have a good, meaningful or supportive answer for this. We'll work with our engineering team to see what (if anything) happened and we will follow up with you in your support case.

>Why have we had no reply in over 2 weeks? This is unacceptable for any paid hosting platform.

The fact that you have gotten a quicker response by leveraging our community is both a blessing and a curse. We recognise that helpdesk response times for customers without SLAs are not currently in line with industry expectations. There is a huge demand for the expertise and insight that my team can offer. We're rapidly looking to scale our service to meet this demand.

>Given the seemingly complete lack of interest from Vercel in providing any support on this matter, why should we continue recommending Vercel to our clients over other options that also support NextJS such as Netlify or Azure SWA?

It is not for the want of trying and I share that with the utmost sincerity. Steps are being taken to ensure that the level of support offered by Vercel is a differentiator for the right reasons.

I wish we were engaging under better circumstances, Lee. Rest assured, this isn't the bar we aim to operate at. I'll get back to you in your support case shortly.
